### DISCUSSION TITLE
Record live and split into .ts segments

### DISCUSSION THREAD

**Discussion by D3XX3R** [streamlink/streamlink#4750] [Q&A] [answered]
I'm trying to record my stream live and I can't seem to find a way to split it into .ts segments easily.
How can I split them with streamlink directly? And if that's not possible, how would I use ffmpeg to help with that?

**Comment by bastimeyer** [comment, marked as answer]
Streamlink doesn't support segmented outputs. It can only output a single progressive stream, which is what's needed when playing back streams in a video player or when writing streams in their entirety to the filesystem. That's the only purpose of Streamlink. If you want to download segmented streams like HLS streams as-is (including playlists / stream manifests), then you'll have to take a look at other tools.

FFmpeg could help you splitting up the concatenated HLS stream again into (different) segments if you pipe the output stream of Streamlink into FFmpeg via `streamlink --stdout ... | ffmpeg -i pipe:0`. You'll have to figure this out on your own though.

**Reply by D3XX3R** [reply]
Let's say I would record not only my own stream but also the stream of my friend. How would I do the piping? Isn't there some sort of naming?

**Reply by bastimeyer** [reply]
https://streamlink.github.io/cli.html#file-output-options
